editors/code/src/config.ts

```diff
@@ -16,10 +18,24 @@ export interface CargoFeatures {
 }
 
 export class Config {
+    readonly raLspServerGithubArtifactName = {
+        linux: "ra_lsp_server-linux",
+        darwin: "ra_lsp_server-mac",
+        win32: "ra_lsp_server-windows.exe",
+        aix: null,
+        android: null,
+        freebsd: null,
+        openbsd: null,
+        sunos: null,
+        cygwin: null,
+        netbsd: null,
+    }[process.platform];
```

Comment:
Not sure about this platform stuff, needs more investigation on which if these ones may use `linux` binary or other.

Reply:
They can't use the Linux binary. Even android can't when you don't link libc statically.

Reply:
Can you use a `switch` here or even a function? It might help less experienced readers worry less about `null` vs. `undefined` and what might happen.

Reply:
Switch is not an expression, but you do can call a function here. Don't know what you mean by `null` vs `undefined` here, there is no default case for the platform, all the possible values of it are encoded in its type definition, so we will always hit one of the variants. Okay, to reduce confusion I'll covert it to private method with switch statement

Reply:
> are encoded in its type definition

:man_facepalming: